src/testsuite/arith128_test_f64.c

```diff
@@ -0,0 +1,2230 @@
+    } else {
+    }
+
+  return (rc);
+}
+#endif
+int
+test_double_cpsgn (void)
+{
+  vf64_t i, j, e;
+  vf64_t k;
+  int rc = 0;
+
+  printf ("\n%s double Copy Sign\n", __FUNCTION__);
+
+  i = (vf64_t) { 0.0, -0.0 };
+  j = (vf64_t) { -0.0, 0.0 };
+  e = (vf64_t) { -0.0, 0.0 };
+  k = vec_copysignf64 (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2f64x ("vec_copysignf64 i=", i);
+  print_v2f64x ("                j=", j);
+  print_v2f64x ("                k=", k);
+#endif
+  rc += check_v2f64x ("vec_copysignf64 1:", k, e);
+
+  i = (vf64_t) { __DBL_MAX__, __DBL_MIN__ };
+  j = (vf64_t) { -0.0, 0.0 };
+  e = (vf64_t) { -(__DBL_MAX__), __DBL_MIN__ };
+  k = vec_copysignf64 (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2f64x ("vec_copysignf64 i=", i);
+  print_v2f64x ("                j=", j);
+  print_v2f64x ("                k=", k);
+#endif
+  rc += check_v2f64x ("vec_copysignf64 2:", k, e);
+
+  i = (vf64_t) { __DBL_EPSILON__, __DBL_DENORM_MIN__ };
+  j = (vf64_t) { -0.0, 0.0 };
+  e = (vf64_t) { -(__DBL_EPSILON__), __DBL_DENORM_MIN__ };
+  k = vec_copysignf64 (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2f64x ("vec_copysignf64 i=", i);
+  print_v2f64x ("                j=", j);
+  print_v2f64x ("                k=", k);
+#endif
+  rc += check_v2f64x ("vec_copysignf64 2:", k, e);
+
+  i = (vf64_t) CONST_VINT128_DW(__DOUBLE_INF, __DOUBLE_NINF);
+  j = (vf64_t) CONST_VINT64_DW(0.0, -0.0);
+  e = (vf64_t) CONST_VINT128_DW(__DOUBLE_INF, __DOUBLE_NINF);
+  k = vec_copysignf64 (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2f64x ("vec_copysignf64 i=", i);
+  print_v2f64x ("                j=", j);
+  print_v2f64x ("                k=", k);
+#endif
+  rc += check_v2f64x ("vec_copysignf64 3:", k, e);
+
+  i = (vf64_t) CONST_VINT128_DW(__DOUBLE_INF, __DOUBLE_NINF);
+  j = (vf64_t) CONST_VINT64_DW(0.0, -0.0);
+  e = (vf64_t) CONST_VINT128_DW(__DOUBLE_INF, __DOUBLE_NINF);
+  k = vec_copysignf64 (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2f64x ("vec_copysignf64 i=", i);
+  print_v2f64x ("                j=", j);
+  print_v2f64x ("                k=", k);
+#endif
+  rc += check_v2f64x ("vec_copysignf64 3:", k, e);
+
+  i = (vf64_t) CONST_VINT128_DW(__DOUBLE_NAN, __DOUBLE_NNAN);
+  j = (vf64_t) CONST_VINT64_DW( -0.0, 0.0 );
+  e = (vf64_t) CONST_VINT128_DW(__DOUBLE_NNAN, __DOUBLE_NAN);
+  k = vec_copysignf64 (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2f64x ("vec_copysignf64 i=", i);
+  print_v2f64x ("                j=", j);
+  print_v2f64x ("                k=", k);
+#endif
+  rc += check_v2f64x ("vec_copysignf64 4:", k, e);
+
+  i = (vf64_t) CONST_VINT128_DW(__DOUBLE_NSNAN, __DOUBLE_SNAN);
+  j = (vf64_t) CONST_VINT64_DW ( 0.0, -0.0 );
+  e = (vf64_t) CONST_VINT128_DW(__DOUBLE_SNAN, __DOUBLE_NSNAN);
+  k = vec_copysignf64 (i, j);
+
+#ifdef __DEBUG_PRINT__
+  print_v2f64x ("vec_copysignf64 i=", i);
+  print_v2f64x ("                j=", j);
+  print_v2f64x ("                k=", k);
+#endif
+  rc += check_v2f64x ("vec_copysignf64 4:", k, e);
```

Comment:
If I understand these `check_v2f64x` calls, the first parameter is emitted as a label, and the last part of the label in this function is an index.  If so, the indexes need updating.  The sequence above is 1, 2, 2, 3, 3, 4, 4.